environments/pendulum.py

```diff
@@ -0,0 +1,70 @@
+from .environments import Environment
+from PIL import Image, ImageDraw
+import numpy as np
+
+
+class Pendulum(Environment):
+
+    """Pendulum System
+
+    Equations of movement are:
+
+        theta'' = -(g/l)*sin(theta)
+
+    """
+
+    def __init__(self, mass, length, p=None, q=None):
+        """Contructor for pendulum system
+
+        Args:
+            mass (float): Pendulum mass
+            length (float): Pendulum length
+            p (float, optional): Generalized momentum. Defaults to None.
+            q ([type], optional): Generalized position. Defaults to None.
```

Comment:
I am not sure I understand the description here: in what sense `p` and `q` are generalized?

Reply:
Ok I understood what they mean here: [Generalized Coordinates](https://en.wikipedia.org/wiki/Generalized_coordinates). I didn't know about this. Should we call them generalized momentum and position only in the abstract `Env` class? Here they represent the cartesian position and the momentum as `m * v`, right?

Reply:
In the case of the pendulum, the generalized coordinates correspond to the phase and angular momentum of the ball

Reply:
Yes, my bad